Simulated Tx with `creationFee` can cause failure due to insufficient funds

Running `make test-sim-profile` on the `develop` gives the following error:

```shell
Simulating... block 18/500, operation 0/405.--- FAIL: BenchmarkSimulation
    simulate.go:300: error on block  18/500, operation (60/296) from x/launch:
        failed to execute message; message index: 0: 3stake is smaller than 7stake: insufficient funds [cosmos/cosmos-sdk@v0.44.6/x/bank/keeper/send.go:187]
        Comment: unable to deliver tx
```

This error can come from the `launch` or `campaign` modules.  Random accounts with enough balance to cover chain or campaign creation fees are chosen, but this can lead to a situation where the account does not have enough funds to cover this fee AND the gas required for the Tx.  This can be fixed by adding a check that the account has the balance to cover the fee + the gas during account selection.

Version: `latest-25bfc55f`

--- a/x/campaign/simulation/simulation.go
+++ b/x/campaign/simulation/simulation.go
@@ -55,6 +55,35 @@ func deliverSimTx(
 	return simulation.GenAndDeliverTxWithRandFees(txCtx)
 }
 
+// deliverSimTx delivers the tx for simulation from the provided message
+func deliverSimTxCustomFee(
+	r *rand.Rand,
+	app *baseapp.BaseApp,
+	ctx sdk.Context,
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	simAccount simtypes.Account,
+	msg TypedMsg,
+	coinsSpent sdk.Coins,
+	customFee sdk.Coins,
+) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+	txCtx := simulation.OperationInput{
+		R:               r,
+		App:             app,
+		TxGen:           simappparams.MakeTestEncodingConfig().TxConfig,
+		Cdc:             nil,
+		Msg:             msg,
+		MsgType:         msg.Type(),
+		Context:         ctx,
+		SimAccount:      simAccount,
+		AccountKeeper:   ak,
+		Bankkeeper:      bk,
+		ModuleName:      types.ModuleName,
+		CoinsSpentInMsg: coinsSpent,
+	}
+	return simulation.GenAndDeliverTx(txCtx, customFee)
+}
+
 // GetCoordSimAccount finds an account associated with a coordinator profile from simulation accounts
 func GetCoordSimAccount(
 	r *rand.Rand,
@@ -267,7 +296,16 @@ func SimulateMsgCreateCampaign(
 
 		// skip if account cannot cover creation fee
 		creationFee := k.CampaignCreationFee(ctx)
-		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee) {
+		customFee, err := simtypes.RandomFees(r, ctx, creationFee)
+		if err != nil {
+			return simtypes.NoOpMsg(
+					types.ModuleName,
+					types.TypeMsgCreateCampaign,
+					"skip campaign creation"),
+				nil,
+				nil
+		}
+		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee.Add(customFee...)) {
 			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateCampaign, "skip campaign creation"), nil, nil
 		}
 
@@ -278,7 +316,7 @@ func SimulateMsgCreateCampaign(
 			sample.Metadata(20),
 		)
 
-		return deliverSimTx(r, app, ctx, ak, bk, simAccount, msg, sdk.NewCoins())
+		return deliverSimTxCustomFee(r, app, ctx, ak, bk, simAccount, msg, sdk.NewCoins(), customFee)
 	}
 }
 

--- a/x/campaign/simulation/simulation.go
+++ b/x/campaign/simulation/simulation.go
@@ -298,12 +298,9 @@ func SimulateMsgCreateCampaign(
 		creationFee := k.CampaignCreationFee(ctx)
 		customFee, err := simtypes.RandomFees(r, ctx, creationFee)
 		if err != nil {
-			return simtypes.NoOpMsg(
-					types.ModuleName,
-					types.TypeMsgCreateCampaign,
-					"skip campaign creation"),
+			return simtypes.OperationMsg{},
 				nil,
-				nil
+				err
 		}
 		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee.Add(customFee...)) {
 			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateCampaign, "skip campaign creation"), nil, nil

--- a/x/launch/simulation/simulation.go
+++ b/x/launch/simulation/simulation.go
@@ -34,11 +34,23 @@ func SimulateMsgCreateChain(ak types.AccountKeeper, bk types.BankKeeper, k keepe
 			// No message if no coordinator
 			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateChain, "skip create a new chain"), nil, nil
 		}
+
 		// skip if account cannot cover creation fee
 		creationFee := k.ChainCreationFee(ctx)
-		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee) {
-			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateChain, "skip campaign creation"), nil, nil
+		customFee, err := simtypes.RandomFees(r, ctx, creationFee)
+		if err != nil {
+			return simtypes.NoOpMsg(
+					types.ModuleName,
+					types.TypeMsgCreateChain,
+					"skip chain creation"),
+				nil,
+				nil
 		}
+
+		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee.Add(customFee...)) {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgCreateChain, "skip chain creation"), nil, nil
+		}
+
 		msg := sample.MsgCreateChain(
 			simAccount.Address.String(),
 			"",
@@ -59,7 +71,7 @@ func SimulateMsgCreateChain(ak types.AccountKeeper, bk types.BankKeeper, k keepe
 			ModuleName:      types.ModuleName,
 			CoinsSpentInMsg: sdk.NewCoins(),
 		}
-		return simulation.GenAndDeliverTxWithRandFees(txCtx)
+		return simulation.GenAndDeliverTx(txCtx, customFee)
 	}
 }
 

--- a/x/launch/simulation/simulation.go
+++ b/x/launch/simulation/simulation.go
@@ -39,12 +39,11 @@ func SimulateMsgCreateChain(ak types.AccountKeeper, bk types.BankKeeper, k keepe
 		creationFee := k.ChainCreationFee(ctx)
 		customFee, err := simtypes.RandomFees(r, ctx, creationFee)
 		if err != nil {
-			return simtypes.NoOpMsg(
-					types.ModuleName,
-					types.TypeMsgCreateChain,
-					"skip chain creation"),
-				nil,
-				nil
+			if err != nil {
+				return simtypes.OperationMsg{},
+					nil,
+					err
+			}
 		}
 
 		if !creationFee.Empty() && !bk.SpendableCoins(ctx, simAccount.Address).IsAllGTE(creationFee.Add(customFee...)) {

